apps/avifenc.c

```diff
@@ -1550,16 +1624,43 @@ int main(int argc, char * argv[])
             } else {
                 input.files[0].settings.autoTiling = boolSettingsEntryOf(AVIF_TRUE);
             }
-        } else if (!strcmp(arg, "--progressive")) {
-            settings.progressive = AVIF_TRUE;
+        } else if (!strcmp(arg, "--auto-progressive")) {
+            settings.autoProgressive = AVIF_TRUE;
+        } else if (!strcmp(arg, "--layer")) {
+            NEXTARG();
+            int layer = atoi(arg);
+            if (layer < 1 || layer > AVIF_MAX_AV1_LAYER_COUNT) {
+                fprintf(stderr, "ERROR: Invalid layer count (valid range [1-%d]): %s\n", AVIF_MAX_AV1_LAYER_COUNT, arg);
+                returnCode = 1;
+                goto cleanup;
+            }
+            settings.layers = layer;
+        } else if (!strcmp(arg, "--scaling-mode") || strpre(arg, "--scaling-mode:")) {
+            avifOptionSuffixType type = parseOptionSuffix(arg, input.filesCount != 0);
+            if (type == AVIF_OPTION_SUFFIX_INVALID) {
+                returnCode = 1;
+                goto cleanup;
+            }
+            NEXTARG();
+            uint32_t frac[2] = { 0, 1 };
+            if (!(parseU32List(frac, 1, arg, '/') || parseU32List(frac, 2, arg, '/')) || frac[0] > INT32_MAX || frac[1] > INT32_MAX) {
+                fprintf(stderr, "ERROR: Invalid scaling mode: %s\n", arg);
+                returnCode = 1;
+                goto cleanup;
+            }
+            if (type == AVIF_OPTION_SUFFIX_UPDATE || input.filesCount == 0) {
```

Comment:
`pendingSettings` will be assigned to `input.files[i].settings.scalingMode` when a input file is encountered, which happens after here. If `|| input.filesCount == 0` is omitted options in `input.files[0]` will be overridden by an empty `pendingSettings`.

Reply:
Then I have the feeling that it would be simpler to understand if `pendingSettings` was replaced by `input.files[input.filesCount].settings` everywhere. But this has nothing to do with this PR.